### ISSUE TITLE
Questions about EFL

### ISSUE THREAD

**Issue by qdd1234** [ModelTC/EOD#9]
作者您好，感谢您精彩的工作。我对于下图展示的EFL的公式有一个疑惑希望您能帮我解答一下：
（1）您提及到对于class imbalance严重的类，gamma值应该要很大，但是同样会带来这一类对于最终loss contributions变少，所以您又加上一个weight factor，但是我在复现您代码的时候有个问题，就是您是分别计算每一类的损失然后求和，那您再计算某一类loss的时候，其他类的gamma值是怎么确定的？直接赋值为0吗？因为在您代码中您是直接乘上一个计算得到的gamma(第二个图)，但是这样对于某一类而言，只是class imbalance类的gamma值非常小但是background类的gamma值却有时比较大，您可以再细化一下这个公式的解释吗，谢谢！
![image](https://user-images.githubusercontent.com/68497831/148882763-fbb8e4ad-ad67-4e1a-ac1f-f7899d9667f5.png)
![image](https://user-images.githubusercontent.com/68497831/148883403-18ac85ce-a764-4e11-b4e0-68e4c47bbcf3.png)

**Comment by waveboo** [comment]
Hi @qdd1234 , thanks for your question.
For the loss function, EFL is generated from the Sigmoid Focal Loss. If there are C foreground categories, there will be C classifiers and each classifier is responsible for a specific category, i.e. a binary classification task.
For the param `dy_gamma` in fig2, we calculate the values for all categories at once (it is a C-dim vector in our code). 
Actually, `self.focal_gamma` is corresponded to the gamma_b, `self.scale_factor` is corresponded to the hyper-param s, and `map_val` is  corresponded to the 1 - g. We statistics g in `self.pod_neg` for all categories (it is also a C-dim vector).
You could run our code of EFL and debug it to obtain more info. The guide is at [EFL](https://github.com/ModelTC/EOD/blob/main/docs/equalized_focal_loss.md)

**Comment by yqyao** [comment]
This issue is closed due to not being active. Please feel free to open it again @qdd1234.
